test/index.test.ts

```diff
@@ -192,6 +192,116 @@ describe('HipThrusTS', () => {
       });
     });
   });
+  describe('Hipthrusts functional only', () => {
+    describe('HTPipeAttachData', async () => {
+      const left = {
+        attachData(context: { a: string }) {
+          return { b: 4 };
+        },
+      };
+      const leftBad = {
+        attachData(context: { a: string }) {
+          return { b: 'bad' };
+        },
+      };
+      const rightFullyCovered = {
+        attachData(context: { b: number }) {
+          return { c: 4 };
+        },
+      };
+      const rightPartiallyCovered = {
+        attachData(context: { b: number; other: string }) {
+          return { c: 4 };
+        },
+      };
+      const rightNotCovered = {
+        attachData(context: { other: string }) {
+          return { c: 4 };
+        },
+      };
+
+      // async paths
+
+      const leftAsync = {
+        attachData(context: { a: string }) {
+          return Promise.resolve({ b: 4 });
+        },
+      };
+      const leftBadAsync = {
+        attachData(context: { a: string }) {
+          return Promise.resolve({ b: 'bad' });
+        },
+      };
+      const rightFullyCoveredAsync = {
+        attachData(context: { b: number }) {
+          return Promise.resolve({ c: 4 });
+        },
+      };
+      const rightPartiallyCoveredAsync = {
+        attachData(context: { b: number; other: string }) {
+          return Promise.resolve({ c: 4 });
+        },
+      };
+      const rightNotCoveredAsync = {
+        attachData(context: { other: string }) {
+          return Promise.resolve({ c: 4 });
+        },
+      };
+
+      // HTPipeAttachData with sync
+
```

Comment:
const pipedError = HTPipeAttachData(leftBad,rightFullyCovered);
it's can't compiled because of type mismatch
i remove it and add type checking so we can be sure that types are good

Reply:
directly check the types of the return values... not just their members.  it's too optimistic otherwise.